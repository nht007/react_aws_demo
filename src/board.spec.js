import React from 'react';
import Board from './board';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  let squares = [
    {x: 0, y: 0, value: 'X'},
    {x: 1, y: 0, value: null},
    {x: 2, y: 0, value: null},
    {x: 0, y: 1, value: 'X'},
    {x: 1, y: 1, value: 'O'},
    {x: 2, y: 1, value: null},
    {x: 0, y: 2, value: null},
    {x: 1, y: 2, value: null},
    {x: 2, y: 2, value: null},
  ]

  const tree = renderer
    .create(
      <Board
        squares={squares}
        onClick={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
