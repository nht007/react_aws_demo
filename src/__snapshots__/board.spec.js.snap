// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="board-row"
  >
    <button
      className="square"
      onClick={[Function]}
    >
      X
    </button>
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <button
      className="square"
      onClick={[Function]}
    >
      X
    </button>
    <button
      className="square"
      onClick={[Function]}
    >
      O
    </button>
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
</div>
`;
